### ISSUE TITLE
default DATA_DIR + path.data merging

### ISSUE THREAD

**Issue by thekad** [elastic/puppet-elasticsearch#584]
Using ES 1.4.2, this is very likely related to #280, briefly:

1. You have an instance, say, ``es01``, single non-default data directory i.e. ``/mnt/data/elasticsearch`` (automatically appends the ``es01`` as per instance)
2. You pass the above setting via ``config`` parameter to the ``elasticsearch::instance`` resource
3. You start your node
4. You check your node's FS stats and it turns out it's got 2 data directories, even though you only configured one

I've tracked down the issue to https://github.com/elastic/puppet-elasticsearch/blob/master/templates/etc/init.d/elasticsearch.Debian.erb#L83 and https://github.com/elastic/elasticsearch/blob/v1.4.2/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java#L60-L64

TL;DR the init config uses ``es.default.path.data``, but setting ``path.data`` in the yml file will *append* to the list of directories (if my understanding of the settings loader is correct)

The workaround would be to set ``DATA_DIR`` in the ``init_defaults`` parameter when you create the instance, because then it gets propagated properly.

Not 100% sure where the bug here is, the documentation clearly states that es.default.* keys are used only when no defaults are set, but the java code suggests otherwise.

**Comment by thekad** [comment]
Not sure why this is still open

**Comment by thekad** [comment]
... because I got confused and this is the upstream issue, duh.

**Comment by tylerjl** [comment]
Hey @thekad, I'm in the process of cleaning up the acceptance test suite and in fixing up the tests, this bug actually resurfaced on all the Debian operating systems that still use SysV init. Your bug sniffing was way ahead of the automated test suite. 😉 

What's interesting to me is that [the SysV script for Debian defines `DATA_DIR`](https://github.com/elastic/puppet-elasticsearch/blob/master/templates/etc/init.d/elasticsearch.Debian.erb#L83) while [the RHEL template](https://github.com/elastic/puppet-elasticsearch/blob/master/templates/etc/init.d/elasticsearch.RedHat.erb) doesn't handle it anywhere. I'm inclined to just make them behave similarly and leave several of the `es.default` options undefined in the Debian template like the RHEL one unless they crop up in the instance's `init_defaults` file, thoughts?

**Comment by thekad** [comment]
Hi @tylerjl,

Yeah I think that sounds fine, having them both behave consistently seems like a good idea to me :)

**Comment by tylerjl** [comment]
One issue that came up in testing this change is that since the `/var/lib` directory isn't being appended to the list of data paths, there's the potential here for the change to happen, peoples' ES gets restarted, and shards are missing part of their data path since `/var/lib` isn't being appended to Debian configs any more.

Is a big disclaimer in the release notes sufficient notice for people, or is this a drastic enough change it needs to be postponed for a major point release? Looking for feedback here from deb-based distro users on SysV init.
